inc/ProgEvent.h

```diff
@@ -163,52 +69,30 @@ namespace PCOE {
         /**
          * Gets the start time of the event.
          **/
-        inline const UData& getStartTime() const {
-            return startTime;
-        }
-
-        /**
-         * Gets the start position of the event.
-         *
-         * @remarks
-         * For events that do not map to a particular location, this property
-         * will generally be set to the origin.
-         **/
-        inline Point3D getStartPosition() const {
-            return startPosition;
+        inline const UData& getTOE() const {
+            return toe;
         }
 
         /**
-         * Gets the end time of the event.
-         *
-         * @remarks
-         * For events that are instantaneous in nature, this property will
-         * generally be the same as the start time.
+         * Gets the set of state associated with the event.
          **/
-        inline const UData& getEndTime() const {
-            return endTime;
+        inline const std::vector<Point4D<MessageClock>> getPoints() const {
+            return points;
         }
 
         /**
-         * Gets the end position of the event.
-         *
-         * @remarks
-         * For events that do not map to a particular location, this property
-         * will generally be set to the origin. For events that are
-         * instantaneous in nature, this property will generally be set to the
-         * start position.
+         * Gets the tag associated with the event.
          **/
-        inline Point3D getEndPosition() const {
-            return endPosition;
+        inline const std::string& getTag() const {
+            return tag;
         }
 
     private:
         MessageId eventId;
         std::vector<UData> eventState;
-        UData startTime;
-        Point3D startPosition;
-        UData endTime;
-        Point3D endPosition;
+        UData toe;
+        std::vector<Point4D<MessageClock>> points;
+        std::string tag;
```

Comment:
Is this different than the event id?

Reply:
I see above that it is “additional information” maybe an example would help me understand

Reply:
For a proximity prediction, there may be several events, each for a different hazard. The event id would be the same for all of them. I'm going to use the tag to attach the type and id of the specific hazard to the event.